### PR TITLE
Generic/AssignmentInCondition: improve sniff code coverage

### PR DESCRIPTION
**Pull request by rodrigoprimo**
## Description

This PR improves the code coverage for the AssignementInCondition sniff. This sniff has some defensive code to protect against parse errors, and those were not exercised by the tests. I followed the instructions on #143 for creating parse error tests.

I also checked if there are new assignment operators added in more recent versions of PHP that are not supported by this sniff, but I did not find anything.

## Suggested changelog entry
Improve AssignementInCondition sniff code coverage

## Related issues/external references

Part of #146

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [x] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php [hunk 1]
```diff
@@ -41,46 +41,54 @@ public function getErrorList()
      * The key of the array should represent the line number and the value
      * should represent the number of warnings that should occur on that line.
      *
+     * @param string $testFile The name of the file being tested.
+     *
      * @return array<int, int>
      */
-    public function getWarningList()
+    public function getWarningList($testFile='AssignmentInConditionUnitTest.1.inc')
```
Comment (jrfnl) [inline src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php:48]:
```suggestion
    public function getWarningList($testFile='')
```

**Review by jrfnl — COMMENTED** [review]
LGTM with only one small remark. Will merge it once that's been fixed up. (Feel free to just amend the existing commit) Thanks @rodrigoprimo !

File: src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php (hunk 1 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php]: I went ahead and amended the existing commit to implement this change.

I had added the name of the first test case file as the default value as I was using [InlineControlStructureUnitTest.php](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/src/Standards/Generic/Tests/ControlStructures/InlineControlStructureUnitTest.php#L33) as reference. Is there a reason we want the name of the first test case file as the default value in InlineControlStructureUnitTest::getErrorList() but not in AssignmentInConditionUnitTest::getWarningList()?

**Comment by rodrigoprimo** [comment]
Thanks for reviewing this PR, @jrfnl. I implemented the change you requested and I believe it is now ready to be merged.

File: src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php (hunk 1 above)
Comment (jrfnl) [reply src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php]: > I had added the name of the first test case file as the default value as I was using [InlineControlStructureUnitTest.php](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/src/Standards/Generic/Tests/ControlStructures/InlineControlStructureUnitTest.php#L33) as reference. Is there a reason we want the name of the first test case file as the default value in InlineControlStructureUnitTest::getErrorList() but not in AssignmentInConditionUnitTest::getWarningList()?

There are actually three reasons:
1. The default value has no value in practice. It is an optional argument which is not [enforced via the abstract functions](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/tests/Standards/AbstractSniffUnitTest.php#L448-L459), but is [passed in all cases](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/tests/Standards/AbstractSniffUnitTest.php#L228-L229), so the default is never used in practice.
2. Setting the default value as file 1 also has an assumption implied, while IMO assumptions have no place in a test suite.
3. Maintainability - one less thing to have to keep in sync and to guard against typos ;-)

Does that help ?

To be honest, it wouldn't be a bad idea to change the default value for `$testFile=...` in all PHPCS native test files which have it to `''` for consistency.

**Review by jrfnl — APPROVED** [review]
LGTM! Thanks @rodrigoprimo.

File: src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php (hunk 1 above)
Comment (jrfnl) [reply src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php]: It's up for grabs, so yes, feel free to submit a PR for that. Please add a link to my comment in the PR description, so it's clear for others where this is coming from.

Comment (rodrigoprimo) [reply src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php]: Here is the PR: https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/168

**Pull request merged**